Samples/ExtendedControlAndMetadata/EyeGazeAndBackgroundSegmentation/CameraKsPropertyHelper/PropertyInquiry.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#include "pch.h"
+#include "PropertyInquiry.h"
+#include "PropertyInquiry.g.cpp"
+#include "KSHelper.h"
+#include <winrt/Windows.Media.Devices.h>
+#include "BasicExtendedPropertyPayload.h"
+#include "BackgroundSegmentationPropertyPayload.h"
+#include "BackgroundSegmentationMaskMetadata.h"
+
+
+namespace winrt::CameraKsPropertyHelper::implementation
+{
+    /// <summary>
+    /// Get the data payload as an IPropertyValue associated with a Get command on an extended camera control
+    /// </summary>
+    /// <param name="controller"></param>
+    /// <param name="controlId"></param>
+    /// <returns></returns>
+    Windows::Foundation::IPropertyValue GetExtendedCameraControlPayload(Windows::Media::Devices::VideoDeviceController const& controller, int controlId)
+    {
+        Windows::Foundation::IPropertyValue propertyValueResult = nullptr;
+
+        KSPROPERTY prop(
+            { 0x1CB79112, 0xC0D2, 0x4213, {0x9C, 0xA6, 0xCD, 0x4F, 0xDB, 0x92, 0x79, 0x72} } /* KSPROPERTYSETID_ExtendedCameraControl */,
+            controlId /* KSPROPERTY_CAMERACONTROL_EXTENDED_* */,
+            0x00000001 /* KSPROPERTY_TYPE_GET */);
+        uint8_t* pProp = reinterpret_cast<uint8_t*>(&prop);
+
+        array_view<uint8_t const> serializedProp = array_view<uint8_t const>(pProp, sizeof(KSPROPERTY));
+        auto getResult = controller.GetDevicePropertyByExtendedId(serializedProp, nullptr);
+        if (getResult.Status() == Windows::Media::Devices::VideoDeviceControllerGetDevicePropertyStatus::NotSupported)
+        {
+            return propertyValueResult;
+        }
+        if (getResult.Status() != Windows::Media::Devices::VideoDeviceControllerGetDevicePropertyStatus::Success)
+        {
+            throw hresult_invalid_argument(L"Could not retrieve device property " + winrt::to_hstring(controlId) + L" with status: " + winrt::to_hstring((int)getResult.Status()));
+        }
+        auto getResultValue = getResult.Value();
+        propertyValueResult = getResultValue.as<Windows::Foundation::IPropertyValue>();
+
+        return propertyValueResult;
+    }
+
+    /// <summary>
+    /// Return the result of a GET command of a supported extended control onto the specified VideoDeviceController
+    /// </summary>
+    /// <param name="controller"></param>
+    /// <param name="extendedControlKind"></param>
+    /// <returns></returns>
+    winrt::CameraKsPropertyHelper::IExtendedPropertyPayload PropertyInquiry::GetExtendedControl(winrt::Windows::Media::Devices::VideoDeviceController const& controller, winrt::CameraKsPropertyHelper::ExtendedControlKind const& extendedControlKind)
+    {
+        auto property = GetExtendedCameraControlPayload(controller, static_cast<int>(extendedControlKind));
```

Comment:
>property [](http://example.com/codeflow?start=13&length=8)

nit: change naming to not match language contructs/keywords
[https://docs.microsoft.com/en-us/cpp/extensions/property-cpp-component-extensions?view=msvc-160](https://docs.microsoft.com/en-us/cpp/extensions/property-cpp-component-extensions?view=msvc-160)
[https://docs.microsoft.com/en-us/cpp/cpp/property-cpp?view=msvc-160](https://docs.microsoft.com/en-us/cpp/cpp/property-cpp?view=msvc-160)